src/components/application_manager/test/commands/mobile/unsubscribe_vehicle_request_test.cc

```diff
@@ -207,10 +214,149 @@ void UnsubscribeVehicleRequestTest::UnsubscribeSuccessfully() {
+
+  EXPECT_CALL(
+      mock_app_manager_,
+      ManageMobileCommand(MobileResultCodeIs(mobile_result::OUT_OF_MEMORY), _));
+
+  command->Run();
+}
+
 TEST_F(UnsubscribeVehicleRequestTest, Run_UnsubscribeData_SUCCESS) {
   UnsubscribeSuccessfully();
 }
 
+TEST_F(UnsubscribeVehicleRequestTest, OnEvent_UnknownEvent_UNSUCCESS) {
+  CommandPtr command(CreateCommand<UnsubscribeVehicleDataRequest>());
+  am::event_engine::Event test_event(hmi_apis::FunctionID::INVALID_ENUM);
+  SmartObject message(smart_objects::SmartType_Map);
+  const hmi_apis::Common_Result::eType hmi_result =
+      hmi_apis::Common_Result::SUCCESS;
+  message[am::strings::params][am::hmi_response::code] = hmi_result;
+  message[am::strings::msg_params][kMsgParamKey] = true;
+  test_event.set_smart_object(message);
+  EXPECT_CALL(*(am::MockMessageHelper::message_helper_mock()),
+              HMIToMobileResult(hmi_result)).Times(0);
+
+  EXPECT_CALL(mock_app_manager_, ManageMobileCommand(_, _)).Times(0);
+
+  command->on_event(test_event);
+}
+
+TEST_F(UnsubscribeVehicleRequestTest,
+       OnEvent_AddAlreadyUnsubscribedVI_SUCCESS) {
+  MessageSharedPtr command_msg(CreateMessage(smart_objects::SmartType_Map));
+  (*command_msg)[am::strings::params][am::strings::connection_key] =
+      kConnectionKey;
+  (*command_msg)[am::strings::msg_params][kMsgParamKey] = true;
+  am::VehicleData vehicle_data;
+  vehicle_data.insert(am::VehicleData::value_type(kMsgParamKey, kVehicleType));
+  EXPECT_CALL(*(am::MockMessageHelper::message_helper_mock()), vehicle_data())
+      .WillOnce(ReturnRef(vehicle_data));
+  CommandPtr command(CreateCommand<UnsubscribeVehicleDataRequest>(command_msg));
+
+  am::ApplicationSet application_set_;
+  MockAppPtr mock_app(CreateMockApp());
+  application_set_.insert(mock_app);
+  DataAccessor<am::ApplicationSet> accessor(application_set_, app_set_lock_);
+  EXPECT_CALL(mock_app_manager_, application(kConnectionKey))
+      .WillOnce(Return(mock_app));
+  EXPECT_CALL(mock_app_manager_, applications()).WillOnce(Return(accessor));
+
+  EXPECT_CALL(*mock_app, IsSubscribedToIVI(kVehicleType))
+      .WillOnce(Return(true))
+      .WillOnce(Return(true));
+  EXPECT_CALL(*mock_app, UnsubscribeFromIVI(kVehicleType))
+      .WillOnce(Return(true));
+
+  EXPECT_CALL(
+      mock_app_manager_,
+      ManageMobileCommand(MobileResultCodeIs(mobile_result::SUCCESS), _));
+
+  command->Run();
+
+  am::event_engine::Event test_event(
+      hmi_apis::FunctionID::VehicleInfo_UnsubscribeVehicleData);
+  SmartObject message(smart_objects::SmartType_Map);
+  const hmi_apis::Common_Result::eType hmi_result =
+      hmi_apis::Common_Result::SUCCESS;
+  const mobile_apis::Result::eType mob_result = mobile_apis::Result::SUCCESS;
+  message[am::strings::params][am::hmi_response::code] = hmi_result;
+  message[am::strings::msg_params][kMsgParamKey] = true;
+  test_event.set_smart_object(message);
+
+  EXPECT_CALL(*(am::MockMessageHelper::message_helper_mock()),
```

Comment:
@pvvasilev @okozlovlux In order to stick to DRY principle you may just clear the expectations in the TearDown()